store.rb

```diff
@@ -118,4 +153,26 @@ def get_node(node)
       raise e
     end
   end
+
+  # immediately update cache if node joins/leaves
+  def watch()
+    return if @zk.nil?
+    @root_watcher ||= @zk.register("/", :only => :child) do |event|
```

Comment:
Actually register should be only called once. Every time you call register, a new subscription is returned. So common pattern is you register once and then within callback you do getChildren with watch => true to re-subscribe.

Alternatively, every time callback is called, you can unsubscribe the object returned by register earlier and then call register to get a new subscription.